doc/en/developer/source/eclipse-guide/index.rst

```diff
@@ -18,6 +18,16 @@ Running and debugging
 Run or debug the class ``org.geoserver.web.Start`` in the ``web-app`` 
 module. The steps to do so are detailed in the :ref:`quickstart`.
 
+Running GeoServer with Extensions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, GeoServer will run without any extensions enabled. In order to run GeoServer with extensions, the ``web-app`` module declares a number of profiles used to enable specific extensions when running ``Start``. To enable an extension, re-generate the ``web-app`` eclipse profile with the appropriate maven profile(s) enabled::
+
+  % cd web/app
+  % mvn eclipse:eclipse -P wps
```

Comment:
I think Andrea is right. If I want to debug an extension I do a mvn eclipse:clean eclipse:eclipse for the extension, import the project in my workspace and add the project to the class path (project settings)  of web-app.